weblate/machinery/forms.py

```diff
@@ -288,6 +299,14 @@ class OpenAIMachineryForm(KeyMachineryForm):
         initial="auto",
         choices=MODEL_CHOICES,
     )
+    custom_model = forms.CharField(
+        label=pgettext_lazy(
+            "Custom model name",
+            "Custom model name",
+        ),
+        help_text=gettext_lazy("Only needed when model is set to 'Custom model'"),
+        required=False,
+    )
```

Comment:
This deserves additional form validation that it is non-blank when custom model choice is selected.